src/database/sqlx-mysql/tests/repositories/test_mysql_account_repository.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{SubsecRound, Utc};
+use database_common::models::{AccountModel, AccountOrigin};
+use database_common::run_transactional;
+use database_sqlx_mysql::{MySqlAccountRepository, MySqlConnectionPool, MySqlPlugin};
+use dill::{Catalog, CatalogBuilder, Component};
+use kamu_core::auth::AccountRepository;
+use sqlx::MySqlPool;
+use uuid::Uuid;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_group::group(database, mysql)]
+#[test_log::test(sqlx::test(migrations = "../migrations/mysql"))]
+async fn test_missing_account_not_found(mysql_pool: MySqlPool) {
+    let harness = MySqlAccountRepositoryHarness::new(mysql_pool);
+
+    run_transactional(&harness.catalog, async move |catalog: Catalog| {
+        let account_repo = catalog.get_one::<dyn AccountRepository>().unwrap();
+
+        let maybe_account = account_repo
+            .find_account_by_email("test@example.com")
+            .await
+            .unwrap();
+
+        assert!(maybe_account.is_none());
+        Ok(())
+    })
+    .await
+    .unwrap();
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[test_group::group(database, mysql)]
+#[test_log::test(sqlx::test(migrations = "../migrations/mysql"))]
+async fn test_insert_and_locate_account(mysql_pool: MySqlPool) {
+    let harness = MySqlAccountRepositoryHarness::new(mysql_pool);
+
+    run_transactional(&harness.catalog, async move |catalog: Catalog| {
```

Comment:
If you are having issues with async lambdas you can try instead of
`async | | { }`
use
`| | async { }`
the latter is a normal (non-async) lambda that returns a future create by `async {}` block.

Reply:
Yeah, it helps. Totally not obvious why one thing works, and another doesn't.

Reply:
Don't quote me on this but I think the complexity of async lambdas feature comes from needing to define async versions of `Fn / FnOnce / FnMut` traits. I didn't yet encounter a situation where async block workaround was not enough.